server/src/data-layer/services/BookingSlotsService.test.ts

```diff
@@ -0,0 +1,79 @@
+import { cleanFirestore } from "test-config/TestUtils"
+import BookingSlotsService from "./BookingSlotsService"
+import { Timestamp } from "firebase-admin/firestore"
+import { BookingSlot } from "data-layer/models/firebase"
+import FirestoreCollections from "data-layer/adapters/FirestoreCollections"
+
+const timestamp = Timestamp.now()
+const bookingSlotData: BookingSlot = {
+  date: timestamp,
+  stripe_product_id: "product_id_value",
+  description: "booking_slot_description",
+  max_bookings: 0
+}
+
+describe("BookingSlotsService Tests", () => {
+  afterEach(async () => {
+    await cleanFirestore()
+  })
+
+  it("Should create a booking slot", async () => {
+    const { id } = await new BookingSlotsService().createBookingSlot(
+      bookingSlotData
+    )
+    const doc = await FirestoreCollections.bookingSlots.doc(id).get()
+    const data = doc.data()
+
+    expect(data).not.toBe(undefined)
+    expect(data).toEqual({ ...bookingSlotData }) // not sure if this is correct
+  })
+
+  it("Should get booking slots based on stripe product id", async () => {
```

Comment:
I would love for us to better test the query here to make sure it works. Here's a scenario I think works well for us:
1. insert `bookingSlotData`
2. insert a random booking slot (that has different data)
3. insert `bookingSlotData` (except the date is the next date)
we could create these three objects before our test handlers (we currently only create one, `bookingSlotData`).

this way the query will test both
- the fact that it will not select the random booking slot
- the fact that it will select _all_ booking slots that have the same Stripe ID.